docs/design/enclave-tls/design.md

```diff
@@ -0,0 +1,364 @@
+	 * By default, the value of type equals to name.
+	 */
+	char type[ENCLAVE_ATTESTER_TYPE_NAME_SIZE];
+	uint8_t priority; // The higher the value, the higher the priority.
+
+	// Detect whether the current attester instance can run in the current environment
+	enclave_attester_err_t (*pre_init)(void);
+	// Initialize the attester instance
+	enclave_attester_err_t (*init)(enclave_attester_ctx_t *ctx, enclave_tls_cert_algo_t algo);
+	// Collect extended certificate materials
+	enclave_attester_err_t (*extend_cert)(enclave_attester_ctx_t *ctx,
+					      const enclave_tls_cert_info_t *cert_info);
+	// Collect evidence for generating Quote
+	enclave_attester_err_t (*collect_evidence)(enclave_attester_ctx_t *ctx,
+						   attestation_evidence_t *evidence,
+						   enclave_tls_cert_algo_t algo, uint8_t *hash);
+	// Clean up attester instance
+	enclave_attester_err_t (*cleanup)(enclave_attester_ctx_t *ctx);
+} enclave_attester_opts_t;
+```
+
+`libattester_<type>.so` will register metadata and methods to the Enclave-TLS core when it is initialized.
+
+```c
+enclave_attester_err_t enclave_attester_register(const enclave_attester_opts_t *opts);
+```
+
+## Enclave Verifier instance
+
+Each Verifier instance is designed with the following metadata and API to verify the received quote data in various formats. The following is the meaning of each metadata and method.
+
+```c
+typedef struct {
+	uint8_t api_version;
+	unsigned long flags;
+	const char name[ENCLAVE_VERIFIER_TYPE_NAME_SIZE];
+	/* Different attester instances may generate the same format of verifier,
+	 * e.g, sgx_ecdsa and sgx_ecdsa_qve both generate the format "sgx_ecdsa".
+	 * By default, the value of type equals to name.
+	 */
+	char type[ENCLAVE_VERIFIER_TYPE_NAME_SIZE];
+	uint8_t priority; // The higher the value, the higher the priority
+
+	// Detect whether the current verifier instance can run in the current environment
+	enclave_verifier_err_t (*pre_init)(void);
+	// Initialize the verifier instance
+	enclave_verifier_err_t (*init)(enclave_verifier_ctx_t *ctx, enclave_tls_cert_algo_t algo);
+	// Verify the received quote evidenc
+	enclave_verifier_err_t (*verify_evidence)(enclave_verifier_ctx_t *ctx,
+						  attestation_evidence_t *evidence, uint8_t *hash,
+						  uint32_t hash_len);
+	// Collect collateral
+	enclave_verifier_err_t (*collect_collateral)(enclave_verifier_ctx_t *ctx);
+	// Clean up verifier instance
+	enclave_verifier_err_t (*cleanup)(enclave_verifier_ctx_t *ctx);
+} enclave_verifier_opts_t;
+```
+
+`libverifier_<type>.so` will register metadata and methods to the Enclave-TLS core when it is initialized.
+
+```c
+enclave_verifier_err_t enclave_verifier_register(const enclave_verifier_opts_t *opts);
+```
+
+## Crypto Wrapper instance
+
+Each Crypto Wrapper instance designs the following metadata and API to cooperate with other instances to complete operations related to cryptographic algorithms. The following is the meaning of each metadata and method.
```

Comment:
Each Crypto Wrapper instance designs the following 
The three instances' description here maybe need to align.